api/main.go

```diff
@@ -77,545 +76,707 @@ func NewApiServer(logger *zap.SugaredLogger, address string, metaAddr string, sr
+		proxyStream, err := serv.meta.GetLabelVariants(ctx)
 		if err != nil {
+			loggerWithResource.Errorw("Failed to get label variants from server", "label variant", nameVariantRequest.NameVariant, "Error", err)
 			return err
 		}
-		sErr := proxyStream.Send(nameVariant)
+		sErr := proxyStream.Send(nameVariantRequest)
 		if sErr != nil {
+			loggerWithResource.Errorw("Failed to send label variants to the server", "label variant", nameVariantRequest.NameVariant, "Error", err)
 			return sErr
 		}
 		res, err := proxyStream.Recv()
 		if err != nil {
+			loggerWithResource.Errorw("Failed to receive label variants from server", "label variant", nameVariantRequest.NameVariant, "Error", err)
 			return err
 		}
 		sendErr := stream.Send(res)
 		if sendErr != nil {
+			loggerWithResource.Errorw("Failed to send label variants to client", "label variant", nameVariantRequest.NameVariant, "Error", err)
 			return sendErr
 		}
 	}
 }
 
 func (serv *MetadataServer) GetSources(stream pb.Api_GetSourcesServer) error {
+	requestID, ctx, logger := serv.Logger.InitializeRequestID(stream.Context())
 	for {
-		name, err := stream.Recv()
+		nameRequest, err := stream.Recv()
+		loggerWithResource := logger.WithResource("source", nameRequest.Name.Name, "")
+		loggerWithResource.Debugw("Get source %v from stream", nameRequest)
+		nameRequest.RequestId = requestID
 		if err == io.EOF {
 			return nil
 		}
 		if err != nil {
-			serv.Logger.Errorf("Failed to read client request: %v", err)
+			loggerWithResource.Errorw("Failed to read client request", "source", nameRequest.Name, "Error", err)
 			return err
 		}
-		proxyStream, err := serv.meta.GetSources(stream.Context())
+		proxyStream, err := serv.meta.GetSources(ctx)
 		if err != nil {
+			loggerWithResource.Errorw("Failed to get sources from server", "source", nameRequest.Name, "Error", err)
 			return err
 		}
-		sErr := proxyStream.Send(name)
+		sErr := proxyStream.Send(nameRequest)
 		if sErr != nil {
+			loggerWithResource.Errorw("Failed to send source to the server", "source", nameRequest.Name, "Error", err)
 			return sErr
 		}
 		res, err := proxyStream.Recv()
 		if err != nil {
+			loggerWithResource.Errorw("Failed to receive sources from server", "source", nameRequest.Name, "Error", err)
 			return err
 		}
 		sendErr := stream.Send(res)
 		if sendErr != nil {
+			loggerWithResource.Errorw("Failed to send sources to client", "source", nameRequest.Name, "Error", err)
 			return sendErr
 		}
 	}
 }
 
 func (serv *MetadataServer) GetSourceVariants(stream pb.Api_GetSourceVariantsServer) error {
+	requestID, ctx, logger := serv.Logger.InitializeRequestID(stream.Context())
 	for {
-		nameVariant, err := stream.Recv()
+		nameVariantRequest, err := stream.Recv()
+		loggerWithResource := logger.WithResource("source", nameVariantRequest.NameVariant.Name, nameVariantRequest.NameVariant.Variant)
+		loggerWithResource.Debugw("Get source variant %v from stream", nameVariantRequest.NameVariant)
```

Comment:
I think these should all be info

Reply:
I chose this to be a debug instead of an info because it is a for loop, so I thought it would clutter the logs. I have put one info at the beginning of the function and debugs in the for loop